server/image_list.go

```diff
@@ -33,16 +33,20 @@ func ConvertImage(from *storage.ImageResult) *types.Image {
 		return nil
 	}
 
-	repoTags := []string{"<none>:<none>"}
+	repoTags := []string{}
 	if len(from.RepoTags) > 0 {
 		repoTags = from.RepoTags
 	} else if from.PreviousName != "" {
 		repoTags = []string{from.PreviousName + ":<none>"}
 	}
 
-	repoDigests := []string{"<none>@<none>"}
-	if len(from.RepoDigests) > 0 {
+	repoDigests := []string{}
+	if len(from.RepoDigests) > 0 { // nolint: gocritic
 		repoDigests = from.RepoDigests
+	} else if from.Name != "" {
+		repoDigests = []string{from.Name + "@" + from.Digest.String()}
```

Comment:
If the same image is pulled from different locations, and those locations have different digests, can’t this return an impossible name@digest combination (after pulling repo1@digest1 and repo2@digest2, returning a repo1@digest2 which does not physically exist)?